### ISSUE TITLE
False positive Unnecessary semicolon (standard:no-semi)

### ISSUE THREAD

**Issue by vanniktech** [pinterest/ktlint#2771]
The following gets flagged by the `no-semi` rule: 

```kotlin
enum class Test {
  ;

  val test = ordinal + 1
}
```

Also the `;` is removed when using the auto format option; however then it is a compile error since the `;` is required. So for enums that contain NO entries but variables or other statements the `;` is required.

**Comment by paul-dingemans** [comment]
Tnx for reporting. Out of interest, what is your use case for using an `enum class` without values?

**Comment by vanniktech** [comment]
I store my Feature Flags in my enums. After a while I purge them when I've activated them and they are working fine. And it just happened to be that right now I don't have any Feature Flags but I want to keep my logic around them (computed properties).
